AICopilot/socket_server.py

```
@@ -308,23 +323,44 @@ def _handle_client(self, client_socket):
-            
+
             # Extract tool name and arguments
             tool_name = command.get('tool')
             args = command.get('args', {})
-            
-            # Route to appropriate handler
-            result = self._execute_tool(tool_name, args)
-            
+
+            # Route to appropriate handler on the main GUI thread
+            def dispatch_task():
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `dispatch_task`

(ANN202)